### DISCUSSION TITLE
useContractWrite onError not work

### DISCUSSION THREAD

**Discussion by ChrisChan888** [wevm/wagmi#1990] [Q&A]
Tx is execution reverted however, onError did not alert the error.
```
  const { write } = useContractWrite({
    ...config,
    async onSuccess(data) {
      window.alert('sent!')
      await data.wait()
      window.alert(`After wait: ${data.hash}`)
    },
    onError(data) {
      window.alert(`Error!${data}`)
    },
  })
```

**Comment by tmm** [comment]
`useContractWrite` `onError` is for failures to send transaction to the network. For reverts, you probably want `useWaitForTransaction` [`onError`](https://wagmi.sh/react/hooks/useWaitForTransaction#onerror-optional).
